src/tests/test_resultset.py

```diff
@@ -635,3 +637,48 @@ def test_doesnt_refresh_sqlaproxy_if_different_connection():
     list(first_set)
 
     assert id(first_set._sqlaproxy) == original_id
+
+
+@pytest.mark.parametrize(
+    "function, expected_warning",
+    [
+        (
+            "pie",
+            (
+                ".pie() is deprecated and will be removed in a future version. "
+                "Use %sqlplot pie instead. "
+                "For more help, find us at https://ploomber.io/community "
+            ),
+        ),
+        (
+            "bar",
+            (
+                ".bar() is deprecated and will be removed in a future version. "
+                "Use %sqlplot bar instead. "
+                "For more help, find us at https://ploomber.io/community "
+            ),
+        ),
+        (
+            "plot",
+            (
+                ".plot() is deprecated and will be removed in a future version. "
+                "For more help, find us at https://ploomber.io/community "
+            ),
+        ),
+    ],
+)
+def test_deprecated_warnings(config, function, expected_warning):
+    with warnings.catch_warnings(record=True) as record:
+        if function == "pie":
+            df = pd.DataFrame({"x": [1, 2, 3], "y": [4, 5, 6]})  # noqa
+        else:
+            df = pd.DataFrame({"x": [1, 2, 3]})  # noqa
+
+        engine = sqlalchemy.create_engine("duckdb://")
+        conn = SQLAlchemyConnection(engine)
+        result = conn.raw_execute("select * from df")
+        rs = ResultSet(result, config, statement="select * from df", conn=conn)
+
+        getattr(rs, function)()
```

Comment:
i'm confused about this test. looks like this line is the one that should be triggering the warning (because this is what calls the plotting function), so why is this outside the catch warnings context manager? even more confusing: how come this test is passing if the context manager only has a data frame declaration?

Reply:
Restructured the function so that the catch warnings context manager only contains the line that should be triggering the warning. This test should only pass if one warning is emitted and the warning matches the expected text.